x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/ConnectorIndexService.java

```diff
@@ -194,13 +242,19 @@ private Connector createConnectorWithDefaultValues(
      * Gets the {@link Connector} from the underlying index.
      *
      * @param connectorId The id of the connector object.
+     * @param isDeleted   If set to true, it returns only soft-deleted connector; otherwise, it returns non-deleted connector.
      * @param listener    The action listener to invoke on response/failure.
      */
-    public void getConnector(String connectorId, ActionListener<ConnectorSearchResult> listener) {
+    public void getConnector(String connectorId, Boolean isDeleted, ActionListener<ConnectorSearchResult> listener) {
```

Comment:
One interesting side effect of it is that it's possible to have a non-deleted and deleted connectors with same id that will actually point to really different connectors. That can cause some confusion, but probably nothing flow-breaking, other than customer restoring deleted connector manually breaking something, but that's on them? Should we document this more?

Reply:
Also in same scenario, deleting a live connector would erase previous record of a deleted connector. I guess it's also fine given our limitations, but might be nice to document.